CodeCoverageQualityGate/CodeCoverageQualityService.ts

```diff
@@ -170,32 +171,32 @@ export class CodeCoverageQualityService {
         return coverageInfo;
     };
 
-    private getCoberturaReportOfCurrentBuild = async (): Promise<FileEntry | undefined> => {
+    private getMergedCoberturaReportOfCurrentBuild = async (): Promise<FileEntry | undefined> => {
         const currentBuildArtifact = await this.buildClient.getCoberturaArtifactOfBuildById(Number(this.buildId));
         if (!currentBuildArtifact) {
             return;
         }
-        const coberturaReports = await this.buildClient.getCoberturaReportsOfArtifact(currentBuildArtifact);
+        const coberturaReports = (await this.buildClient.getMergedCoberturaReportsOfArtifact(currentBuildArtifact)).filter((file) => file.name.includes(this.MERGED_COBERTURA_REPORT_NAME));
         if (coberturaReports.length == 0) {
             return;
         }
-        // Assuming there is only one Cobertura report in artifact
+        // Using first report When multiple merged Cobertura reports or only one merged Cobertura report exist
```

Comment:
Instead of using the first report here, can we change the method `getMergedCoberturaReportsOfArtifact` to return only one report?